### PR TITLE
fix(batch): add missing BatchActionType

### PR DESCRIPTION
**Pull request by bodinsamuel**
Ref: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cdffe8978f1bb97cba1e8eddb30580dd08456a54/types/algoliasearch/index.d.ts#L747-L754

### CODE REVIEW

**Comment by Haroenv** [comment]
Seems to be inconsistent with the docs: https://www.algolia.com/doc/api-reference/api-methods/batch/#method-param-action

Do the docs also need to be updated?

File: packages/client-search/src/types/BatchActionType.ts [hunk 1]
```diff
@@ -4,11 +4,15 @@ export const BatchActionEnum: Readonly<Record<string, BatchActionType>> = {
   PartialUpdateObject: 'partialUpdateObject',
   PartialUpdateObjectNoCreate: 'partialUpdateObjectNoCreate',
   DeleteObject: 'deleteObject',
+  DeleteIndex: 'delete',
```
Comment (nunomaduro) [inline packages/client-search/src/types/BatchActionType.ts:7]: @Ant-hem @aseure can you quick check those in your clients?

**Comment by bodinsamuel** [comment]
For the ref, we are currently using that in the crawler so it works. Probably need a doc update indeed.

Example:
```js
var algoliasearch = require("algoliasearch");

const client = algoliasearch('<app>', '<key>');
await client.multipleBatch([
    {
      indexName: 'test-multiple-delete',
      action: 'addObject',
      body: { objectID: 'sample-objectID' },
    },
    {
      indexName: 'test-multiple-delete',
      action: 'clear',
      body: {},
    },
    {
      indexName: 'test-multiple-delete',
      action: 'delete',
      body: {},
    },
  ]);
```

File: packages/client-search/src/types/BatchActionType.ts (hunk 1 above)
Comment (aseure) [reply packages/client-search/src/types/BatchActionType.ts:7]: What do we need to check? If batch actions do accept `delete`? Could you open issues with `polyglot` with the exact context?

Comment (nunomaduro) [reply packages/client-search/src/types/BatchActionType.ts:7]: @aseure Correct. I will do it.

Comment (aseure) [reply packages/client-search/src/types/BatchActionType.ts:7]: @nunomaduro I've just checked and it's available in Go, Scala, Java, C#.

Comment (bodinsamuel) [reply packages/client-search/src/types/BatchActionType.ts:7]: can we merge this one as everybody seems to agree ? ☺️

**Comment by bodinsamuel** [comment]
@Haroenv can review it during the absence of nuno ? ☺️

**Pull request merged**